core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -1021,4 +1021,19 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties with Logging {
     secretKeys.foreach { key => assert(redactedConf(key) === Utils.REDACTION_REPLACEMENT_TEXT) }
     assert(redactedConf("spark.regular.property") === "not_a_secret")
   }
+
+  test("SparkUrlStreamHandlerFactory") {
+    URL.setURLStreamHandlerFactory(new SparkUrlStreamHandlerFactory())
+
+    // if 'hdfs' is not supported, MalformedURLException will be thrown
+    new URL("hdfs://docs.oracle.com/test.jar")
```

Comment:
you should check to see what happens when you run this test on a machine with no network connection. Everyone hates tests that fail when they rely on DNS working (or, in some cases, DNS not resolving an example.org domain)